Error running make when cloning and updating tesseract -   undefined reference to `__imp_crc32' and [Makefile:4375: tesseract.exe] Error 1

<!---
If you use the built-in options and it's broken because of a lib you didn't even know existed or will never use, please choose your own libs and avoid creating issues about it. I barely have any time or will to maintain this.

For new libs and tools pull requests are strongly recommended instead of opening a feature request for the same reason above.

Make sure to mention if you are building for 32-bit or 64-bit or both and mention what the suite failed to build. Please make sure to at least give the title a little detail as to what you failed to compile.

Any issues without a `logs.zip` or at least an error report with either be ignored or closed without warning. Any issue not dealing with the suite or a custom build will not be supported and should be taken to the proper forums or repo page.

If you understood the above, add the info below:
--->

``` bash
leptonica git  .............................................. [Up-to-date]
  Running git clone for tesseract ...
  Running git update for tesseract ...
┌ tesseract git  ...................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$archive_read_format_7zip_read_data+0x133): undefined reference to `__imp_crc32'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$header_bytes+0x6d): undefined reference to `__imp_crc32'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$header_bytes.constprop.0+0x77): undefined reference to `__imp_crc32'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$parse_7zip_uint64+0x42): undefined reference to `__imp_crc32'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$read_Bools+0x36): undefined reference to `__imp_crc32'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libarchive.a(archive_read_support_format_7zip.c.obj):(.text$archive_read_format_7zip_read_header+0xd0): more undefined references to `__imp_crc32' follow
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:4375: tesseract.exe] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:7983: all-recursive] Error 1
make failed. Check D:/MABS/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach D:\MABS\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.
```

[logs.zip](https://github.com/user-attachments/files/27063125/logs.zip)

[ab-suite.make.log](https://github.com/user-attachments/files/27063152/ab-suite.make.log)

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -2182,7 +2182,9 @@ _check=(bin-video/uvg266.exe libuvg266.a uvg266.pc uvg266/uvg266.h)
 if [[ $bits = 64bit && $uvg266 = y ]] &&
     do_vcs "$SOURCE_REPO_UVG266"; then
     do_uninstall include/uvg266 "${_check[@]}"
-    do_cmakeinstall video -DBUILD_TESTING=OFF
+    grep_or_sed __MINGW32__ src/uvg266.h 's;UVG_STATIC_LIB.*;& || defined(__MINGW32__);'
+    # -DBUILD_SHARED_LIBS is now ignored, have to set it manually
+    do_cmakeinstall video -DUVG_BUILD_TESTS=OFF -DUVG_BUILD_SHARED=OFF -DUVG_BUILD_STATIC=on
     do_checkIfExist
 fi
 

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -637,6 +637,9 @@ if { { [[ $ffmpeg != no || $standalone = y ]] && enabled libtesseract; } ||
         else
             extracommands+=("-Dtiff-tools=OFF")
         fi
+        sed -i 's;message(FATAL_ERROR "The imported target;message(WARNING "The imported target;' \
+        "$MINGW_PREFIX"/lib/cmake/libjpeg-turbo/libjpeg-turboTargets.cmake
+        sed -ri 's/libjpeg-turbo::(|turbo)jpeg/&-static/' cmake/JPEGCodec.cmake
         grep_or_sed 'Requires.private' libtiff-4.pc.in \
             '/Libs:/ a\Requires.private: libjpeg liblzma zlib libzstd glut'
         LDFLAGS+=" $($PKG_CONFIG --libs zlib)" CFLAGS+=" -DFREEGLUT_STATIC $($PKG_CONFIG --cflags zlib)" \

--- a/build/media-suite_helper.sh
+++ b/build/media-suite_helper.sh
@@ -1920,7 +1920,7 @@ fix_impsyms() (
     fi
 
     # 2. Resolve provider library paths from pkg-config
-    mapfile -t pkg_libs < <($PKG_CONFIG --libs "$pkg_name" 2> /dev/null | tr ' ' '\n' | grep -- '.')
+    mapfile -t pkg_libs < <($PKG_CONFIG --libs --keep-system-libs "$pkg_name" 2> /dev/null | tr ' ' '\n' | grep -- '.')
 
     search_paths=()
     lib_names=()

